### DISCUSSION TITLE
Interesting observation with command line switches for 'thorium.bat'

### DISCUSSION THREAD

**Discussion by Girofox** [Alex313031/thorium#568] [Ideas]
I edited and experimented with the file `thorium.bat` and added more flags for my own testing. For that the site https://peter.sh/experiments/chromium-command-line-switches/ was very helpful.

### Additional command line switches

`--disk-cache-dir=nul
--webgl-antialiasing-mode=none
--disable-canvas-aa --disable-2d-canvas-clip-aa --disable-features=OptimizationHints --disable-ipc-flooding-protection`

### Explanation

- `--disk-cache-dir=nul` : (Windows) disables writes to "(Code) Cache" folders in user dir (Linux: `--disk-cache-dir=/dev/null`)
- `--webgl-antialiasing-mode=none` disable webgl antialiasing (probably unnecessary if flag `#webgl-msaa-sample-count` is set to 0)
- `--disable-canvas-aa` : disable antialiasing on 2D canvas (probably unnecessary if flag #webgl-msaa-sample-count is set to 0)
- `--disable-2d-canvas-clip-aa` (probably unnecessary if switch `--disable-canvas-aa` is used)
- `--disable-features=OptimizationHints` : disables Optimization Guide Prediction models (less disk writes and faster startup)
- `--disable-ipc-flooding-protection` : disables the IPC flooding protection. Some javascript functions can be used to flood the browser process with IPC. This protection limits the rate at which they can be used.

It feels that with the last one `--disable-ipc-flooding-protection` thorium (and chrome, edge) feels snappier.  Don't know to properly benchmark this, but should affect everything javascript related.

**Comment by midzer** [comment]
Thanks for sharing your findings.

I've tried `--disable-ipc-flooding-protection`. Reduced memory usage of invidual tabs up to 25%.

**Reply by Girofox** [reply]
Wow, that is very significant! Maybe this is a Chromium bug, no way this should be intended.

**Comment by Banderi** [comment]
Where are these switches parsed in the source code? I can't find it anywhere no matter how hard I look. Some of the source files refer to inclusions that are missing in the repo, like `content/shell/common/shell_switches.h` ...
